### PR TITLE
Pv2 1084 remove last year price episodes from earnings

### PR DESCRIPTION
**Pull request by kuzinator**

### CODE REVIEW

File: src/SFA.DAS.Payments.EarningEvents.Application.UnitTests/Mapping/MappingTest.cs [hunk 1]
```diff
@@ -561,6 +564,206 @@ public void TestMultiplePriceEpisodes()
...
+                    TNP1 = 7500,
+                    TNP2 = 7500,
+                    PriceEpisodeCompleted = true,
+                },
+                PriceEpisodePeriodisedValues = new List<PriceEpisodePeriodisedValues>
+                {
+                    new PriceEpisodePeriodisedValues
+                    {
+                        AttributeName = "PriceEpisodeOnProgPayment",
+                        Period1 = 500,
```
Comment (carlbm) [inline src/SFA.DAS.Payments.EarningEvents.Application.UnitTests/Mapping/MappingTest.cs]: Last year (and future year) prices will always be 0

File: src/SFA.DAS.Payments.EarningEvents.Application.UnitTests/Mapping/MappingTest.cs [hunk 2]
```diff
@@ -561,6 +564,206 @@ public void TestMultiplePriceEpisodes()
...
+                        Period8 = 0,
+                        Period9 = 0,
+                        Period10 = 0,
+                        Period11 = 0,
+                        Period12 = 0
+                    },
+                    new PriceEpisodePeriodisedValues
+                    {
+                        AttributeName = "PriceEpisodeCompletionPayment",
+                        Period1 = 100,
```
Comment (carlbm) [inline src/SFA.DAS.Payments.EarningEvents.Application.UnitTests/Mapping/MappingTest.cs]: Last year (and future year) prices will always be 0

File: src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs [hunk 3]
```diff
@@ -43,14 +43,35 @@ public class IntermediateLearningAim
...
             Ukprn = ukprn;
             AcademicYear = collectionYear;
             CollectionPeriod = collectionPeriod;
             IlrSubmissionDateTime = ilrSubmissionDateTime;
             JobId = jobId;
         }
 
+        private IEnumerable<PriceEpisode> FilterPriceEpisodes(IEnumerable<PriceEpisode> priceEpisodes, short currentAcademicYear)
+        {
+            var calendarYear = currentAcademicYear / 100 + 2000;
```
Comment (carlbm) [inline src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs]: What happens in the 9900 academic year?

File: src/SFA.DAS.Payments.EarningEvents.Application.UnitTests/Mapping/MappingTest.cs [hunk 4]
```diff
@@ -561,6 +564,206 @@ public void TestMultiplePriceEpisodes()
...
+                        Period3 = 0,
+                        Period4 = 0,
+                        Period5 = 0,
+                        Period6 = 0,
+                        Period7 = 0,
+                        Period8 = 0,
+                        Period9 = 0,
+                        Period10 = 0,
+                        Period11 = 0,
+                        Period12 = 500
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.EarningEvents.Application.UnitTests/Mapping/MappingTest.cs]: noted

File: src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs (hunk 3 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs]: Let's wait 80 years and see

Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs]: I think is about time we have an academic year dates service so we can have this logic in one place seems to be everywhere now.

Comment (kuzinator) [reply src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs]: I'd love to bring back CalendarPeriod class with all logic in one place. Separate service to divide and then add - no.

Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs]: We had to create one in commitment and it was very useful. The output will be acaemic year start date and end date. The number of line of codes is irrelevant here, what matters is that we have something that handles a business logic in a central place.

Comment (kuzinator) [reply src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs]: Will consider that when we have one more place in production code that does the same. This particular logic is not used anywhere else apart from test.

Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.EarningEvents.Application/Mapping/IntermediateLearningAim.cs]: Ok fair enough

**Pull request merged**